reporters/GlobalSummaryReporter.js

```diff
@@ -678,20 +680,83 @@ class GlobalSummaryReporter {
       const sender = event.returnValues.from;
       const receiver = event.returnValues.to;
 
+      const isInPeriod = Boolean(
+        event.blockNumber >= this.startBlockNumberForPeriod && event.blockNumber < this.endBlockNumberForPeriod
+      );
+      const isInPrevPeriod = Boolean(
+        event.blockNumber >= this.startBlockNumberForPreviousPeriod &&
+          event.blockNumber < this.startBlockNumberForPeriod
+      );
+
       if (receiver !== ZERO_ADDRESS) {
-        // Add to cumulative holder list.
-        cumulativeTokenHolders[receiver] = true;
+        // Add to token holder list.
+        countAllTokenHolders[receiver] = true;
 
-        // Initialize current holder.
+        // Initialize balance if we have not seen this receiver yet.
         if (!currentTokenHolders[receiver]) {
           currentTokenHolders[receiver] = this.toBN("0");
         }
 
-        // Update balance
+        // Update balance for period.
         currentTokenHolders[receiver] = currentTokenHolders[receiver].add(this.toBN(event.returnValues.value));
+
+        // Since we are searching from oldest to newest block, the receiver account's balance for this period will always
+        // be equal to its cumulative balance.
+        if (isInPrevPeriod) {
+          countPrevPeriodTokenHolders[receiver] = true;
+          prevPeriodTokenHolders[receiver] = currentTokenHolders[receiver];
+        }
+        if (isInPeriod) {
+          countPeriodTokenHolders[receiver] = true;
+          periodTokenHolders[receiver] = currentTokenHolders[receiver];
+        }
       }
 
       if (sender !== ZERO_ADDRESS) {
+        // Since we are searching from oldest to newest block, it is possible that the sender has not been seen yet
+        // as a receiver despite it having a balance. So, we need to initialize the sender's balance for this period
+        // before we update its cumulative balance.
+        if (isInPrevPeriod) {
```

Comment:
I am trying to think of a better way to refactor this since we are re-using a lot of the same code to count token holder stats for a specific period within the larger range of [firstTransferEvent, lastTransferEvent].

We want to get stats for the period and the prevPeriod, but only go through all transfer events once.